plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/telemetry/CodeWhispererTelemetryService.kt

```diff
@@ -96,6 +104,18 @@ class CodeWhispererTelemetryService(private val cs: CoroutineScope) {
                 "total session display time: ${CodeWhispererInvocationStatus.getInstance().completionShownTime?.let { Duration.between(it, Instant.now()) }
                     ?.toMillis()?.toDouble()}"
         }
+        var diffDiagnostics = DiagnosticDifferences(
+            added = emptyList(),
+            removed = emptyList()
+        )
+
+        if (isInternalUser(getStartUrl(project))) {
+            val oldDiagnostics = sessionContext.diagnostics.orEmpty()
+            // wait for the IDE itself to update its diagnostics for current file
+            delay(500)
```

Comment:
this will reflect in metrics

Reply:
what if diagnostics run takes longer than 500ms?

Reply:
There is no callback provided by JetBrains for "diagnostic updated complete" once the acceptance is done. Meanwhile, user can also do quick keyboard input immediately after acceptance. 

We ack two issues with the implementation:
1. If IDE itself does not provide diagnostics (like VSC), we will return [] in diagnostics.
2. If IDE cannot finish within 500ms, we will return [] in diagnostics.

There is no workaround for #1 because we cannot force user to install the LSP for that language.
For #2, 500ms is a best middle ground between "too fast to see any diagnostic showing up" and "too slow to be interrupted by next user input".